mongos (at least as compiled by gcc) requires transaction_participant_resource_yielder.cpp
which is defined by mongo_cc_library(name = "transaction" ...) in src/mongo/db/transaction/BUILD.bazel

diff '--color=auto' -ur a/src/mongo/s/BUILD.bazel b/src/mongo/s/BUILD.bazel
--- a/src/mongo/s/BUILD.bazel	2026-09-11 15:16:39.000000000 -0400
+++ b/src/mongo/s/BUILD.bazel	2026-09-17 14:55:54.059825241 -0400
@@ -726,6 +726,7 @@
         "//src/mongo/db/session:session_catalog",
         "//src/mongo/db:startup_warnings_common",
         "//src/mongo/db/stats:system_buckets_metrics",
+        "//src/mongo/db/transaction",
         "//src/mongo/idl:cluster_server_parameter_refresher",
         "//src/mongo/transport:ingress_handshake_metrics",
         "//src/mongo/transport:service_executor",
